streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -122,11 +122,12 @@ public final ThreadCache cache() {
 
     /**
      * @throws ProcessorStateException if there is an error while closing the state manager
+     * @param writeCheckpoint boolean indicating if a checkpoint file should be written
      */
-    void closeStateManager() {
+    void closeStateManager(final boolean writeCheckpoint) {
```

Comment:
I'd rather not overload the Map to have 2 meanings. I think it is more obvious to pass the parameter